guvnor-asset-mgmt/guvnor-asset-mgmt-client/src/main/java/org/guvnor/asset/management/client/editors/repository/wizard/CreateRepositoryWizard.java

```diff
@@ -144,19 +157,26 @@ public int getPreferredWidth() {
     @Override
     public void isComplete( final Callback<Boolean> callback ) {
 
-        callback.callback( true );
+        final int[] unCompletedPages = {this.pages.size()};
+        final boolean[] completed = {false};
 
         //only when all pages are complete we can say the wizard is complete.
         for ( WizardPage page : this.pages ) {
             page.isComplete( new Callback<Boolean>() {
                 @Override
                 public void callback( final Boolean result ) {
-                    if ( Boolean.FALSE.equals( result ) ) {
-                        callback.callback( false );
+
+                    if ( Boolean.TRUE.equals( result ) ) {
+                        unCompletedPages[0]--;
+                        if ( unCompletedPages[0] == 0 ) {
+                            completed[0] = true;
+                        }
                     }
                 }
             } );
         }
+
+        callback.callback( completed[0] );
```

Comment:
@wmedvede OK, I've looked at the pages' implementations too now and I take back the above... as I can see the ```isPageComplete``` methods do not perform any asynchronous operations. I know ```isComplete``` is invoked in response to the ```WizardPageStatusChangeEvent``` which I see you raise in the actual asynchronous validations. All good stuff :)